HerPublicWebsite.BusinessLogic/Services/RegularJobs/RegularJobsService.cs

```diff
@@ -45,4 +56,48 @@ public async Task RunNightlyTasksAsync()
             await dataProvider.PersistAllChangesAsync();
         }
     }
+
+    public async Task<IList<ReferralRequest>> GetReferralsPassedTenWorkingDayThresholdWithNoFollowUp()
+    {
+        var startDate = await AddWorkingDaysToDateTime(DateTime.Now, -10);
+        var newReferrals = await dataProvider.GetReferralRequestsWithNoFollowUpAfterDate(startDate);
+        return newReferrals;
+    }
+
+    public async Task<DateTime> AddWorkingDaysToDateTime(DateTime initialDateTime, int workingDaysToAdd)
+    {
+        int direction = workingDaysToAdd < 0 ? -1 : 1;
+        List<DateTime> holidays = await getHolidays();
+        DateTime newDateTime = initialDateTime;
+        while (workingDaysToAdd != 0)
+            {
+            newDateTime = newDateTime.AddDays(direction);
+            if (newDateTime.DayOfWeek != DayOfWeek.Saturday && 
+                newDateTime.DayOfWeek != DayOfWeek.Sunday && 
+                !holidays.Contains(newDateTime))
+            {
+                workingDaysToAdd -= direction;
+            }
+        }
+        return newDateTime;
+    }
+
+    private async Task<List<DateTime>> getHolidays() {
+            var parameters = new RequestParameters
+            {
+                BaseAddress = "https://www.gov.uk/bank-holidays.json",
+            };
+            var holidayDataJson = await HttpRequestHelper.SendGetRequestAsync<Dictionary<string, Holidays>>(parameters);
+            var englandAndWalesHolidays = holidayDataJson["england-and-wales"];
+            return englandAndWalesHolidays.events.Select(x => x.date).ToList();
```

Comment:
It's more typical in lambdas to use a letter related to the thing being iterated over, so probably `e => …` for `Event`, rather than `x => …`.